single-pool/js/packages/legacy/src/internal.ts

```diff
@@ -0,0 +1,71 @@
+import { Connection, Transaction, TransactionInstruction, PublicKey } from '@solana/web3.js';
+import { Buffer } from 'buffer';
+
+export function rpc(connection: Connection) {
+  return {
+    getAccountInfo(address: string) {
+      return {
+        async send() {
+          const pubkey = new PublicKey(address);
+          return await connection.getAccountInfo(pubkey);
+        },
+      };
+    },
+    getMinimumBalanceForRentExemption(size: bigint) {
+      return {
+        async send() {
+          return BigInt(await connection.getMinimumBalanceForRentExemption(Number(size)));
+        },
+      };
+    },
+    getStakeMinimumDelegation() {
+      return {
+        async send() {
+          const minimumDelegation = await connection.getStakeMinimumDelegation();
+          return { value: BigInt(minimumDelegation.value) };
+        },
+      };
+    },
+  };
+}
+
+export function modernInstructionToLegacy(modernInstruction: any): TransactionInstruction {
+  const keys = [];
+  for (const account of modernInstruction.accounts) {
+    keys.push({
+      pubkey: new PublicKey(account.address),
+      isSigner: !!(account.role & 2),
+      isWritable: !!(account.role & 1),
```

Comment:
You could use `isSignerRole()` and `isWritableRole()` from `@solana/instructions` here.

Reply:
this is the legacy package